contracts/deploy/Create3.sol

```diff
@@ -21,13 +21,28 @@ contract CreateDeployer {
     }
 }
 
+library AddressUtils {
+    function isContract(address _address) internal view returns (bool) {
+        bytes32 existingCodeHash = _address.codehash;
+
+        // https://eips.ethereum.org/EIPS/eip-1052
+        // keccak256('') == 0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470
+        return
+            existingCodeHash != bytes32(0) &&
+            existingCodeHash != 0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470;
+    }
+}
+
 library Create3 {
+    using AddressUtils for address;
     bytes32 internal constant DEPLOYER_BYTECODE_HASH = keccak256(type(CreateDeployer).creationCode);
 
     function deploy(bytes32 salt, bytes memory bytecode) internal returns (address deployed) {
         deployed = deployedAddress(salt, address(this));
 
-        if (deployed.codehash != bytes32(0)) revert AlreadyDeployed();
+        // https://eips.ethereum.org/EIPS/eip-1052
+        // keccak256('') == 0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470
```

Comment:
I think we can remove this comment as `0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470` moved into the library and irrelevant in this context.

Reply:
Good point